### PR TITLE
feat(apigatewayv2-alpha): support WebSockets to State Machine direct integration

### LINKED ISSUE #27164: Amazon.CDK.AWS.Apigatewayv2.Alpha: Support "AWS" as the IntegrationType for WebSockets
### Describe the feature

Currently, `Amazon.CDK.AWS.Apigatewayv2.Alpha` lacks support for `AWS` option as the [IntegrationType](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-integrationtype) for **WebSocket** Apigateway.

The quote from the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-integrationtype):
>`AWS`: for integrating the route or method request with an AWS service action, including the Lambda function-invoking action. With the Lambda function-invoking action, this is referred to as the Lambda custom integration. With any other AWS service action, this is known as AWS integration. Supported only for WebSocket APIs.

The important part is:
>With any other AWS service action, this is known as AWS integration. Supported only for WebSocket APIs.

That is, you can create a WebSocket Apigateway that calls **directly** other AWS services **without** a Lambda function middleware.

### Use Case

Ability to create a CDK for WebSocket Apigateway where I can define a direct integration to another AWS service.

For example, WebSocket route integration that triggers a AWS Step Function directly.

This use case is supported using CloudFormation and SAM as follows:
```yaml
  ApiAuthenticateRouteIntegration:
    Type: AWS::ApiGatewayV2::Integration
    Properties:
      ApiId: !Ref BrokerRobotSocketApi
      IntegrationType: AWS
      IntegrationMethod: POST
      IntegrationUri: !Sub "arn:aws:apigateway:${AWS::Region}:states:action/StartExecution"
      CredentialsArn: !Sub "${ApiAuthenticateIntegrationStateMachineExecutionRole.Arn}" 
      TemplateSelectionExpression: \$default
      RequestTemplates: # see: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-mapping-template-reference.html
        "$default" : 
          Fn::Sub: >
            #set($statesInput=$util.escapeJavaScript($input.json('$.data')).replaceAll("\\'","'"))
            { 
              "input": "{\"data\":$statesInput, \"metadata\": {\"requestId\":\"$context.requestId\",\"connectionId\":\"$context.connectionId\",\"domainName\":\"$context.domainName\",\"stage\":\"$context.stage\"}}",
              "stateMachineArn": "${BrokerAuthenticatorStateMachine}"
            }
```

### Proposed Solution

From dotnet world, the solution requires the creation of another class that inherits from [`WebSocketRouteIntegration`](https://docs.aws.amazon.com/cdk/api/v2/dotnet/api/Amazon.CDK.AWS.Apigatewayv2.Alpha.WebSocketRouteIntegration.html) to expose the specific props needed for IntegrationType as AWS.

I propose a class named `WebSocketAwsIntegration`, following the convention for the other integrations types already implemented: [WebSocketMockIntegration](https://docs.aws.amazon.com/cdk/api/v2/dotnet/api/Amazon.CDK.AWS.Apigatewayv2.Integrations.Alpha.WebSocketMockIntegration.html), [WebSocketLambdaIntegration](https://docs.aws.amazon.com/cdk/api/v2/dotnet/api/Amazon.CDK.AWS.Apigatewayv2.Integrations.Alpha.WebSocketLambdaIntegration.html).

**Alternatively**, we could be even more specific and create classes such as `WebSocketStepFunctionIntegration`, however, this would require considerably more work because we would need to create several classes for _every_ AWS service that is supported with this construct.

In any case, this would require the following changes - _looking briefly at the code_:

### [`WebSocketRouteIntegrationConfig`](https://github.com/aws/aws-cdk/blob/dc87d28110403f6223ed2ee506c35bc79fc7835e/packages/%40aws-cdk/aws-apigatewayv2-alpha/lib/websocket/integration.ts#L130)

Adding the following missing properties:  [IntegrationMethod](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-integrationmethod), [RequestTemplates](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-requesttemplates), [CredentialsArn](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-credentialsarn), [TemplateSelectionExpression](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-integration.html#cfn-apigatewayv2-integration-templateselectionexpression), and probably others depending on the AWS service you want to integrate.

### [`WebSocketIntegrationType`](https://github.com/aws/aws-cdk/blob/dc87d28110403f6223ed2ee506c35bc79fc7835e/packages/%40aws-cdk/aws-apigatewayv2-alpha/lib/websocket/integration.ts#L19)

Adding the following option to the enum: `AWS`.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

v2

### Environment details (OS name and version, etc.)

Windows 11

### PR DESCRIPTION
**Pull request by JobaDiniz**
This is a _draft_ attempt to start implementing the use case described in #27164 and discussed on https://github.com/aws/aws-cdk/discussions/27073.

Since this is my first contribution, I'm seeking help to design and implement this feature. I'm open to restart everything if it's the wrong design 😄 .

Closes #27164.

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/lib/websocket/state-machine.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      principal: new ServicePrincipal('apigateway.amazonaws.com'),
+      sourceArn: Stack.of(route).formatArn({
+        service: 'execute-api',
+        resource: route.webSocketApi.apiId,
+        resourceName: '',
+      }),
+    });
+    */
+
+    //TODO: is this correct?
```
Comment (JobaDiniz) [inline packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/lib/websocket/state-machine.ts:71]: @peterwoodworth would you be able to help answer some of the questions I have to implement this feature?

File: packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/lib/websocket/state-machine.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private readonly stateMachine: IStateMachine,
+    private readonly props: WebSocketStateMachineIntegrationProps = {}) {
+    super(id);
+    this._id = id;
+  }
+
+  bind(options: WebSocketRouteIntegrationBindOptions): WebSocketRouteIntegrationConfig {
+    var route = options.route;
+
+    //TODO: how to setup the CredentialsArn?
```
[the remark this answers is not in the document]
Comment (peterwoodworth) [reply packages/@aws-cdk/aws-apigatewayv2-integrations-alpha/lib/websocket/state-machine.ts:58]: Hey @JobaDiniz, thanks a bunch for getting this draft started. I'm not the right one to help with the specifics on this, I'll reach out to find someone who can help with a thorough review on this 🙂

**Comment by scanlonp** [comment]
Hi @JobaDiniz, thanks for opening this pull! I like the look of what you proposed in your feature request issue. The most effective way for us to help you with design will be to edit the readme with what you expect the user experience to be. I saw you had a couple different ideas on how the classes should be set up. Try to keep the end user in mind and how you / they would want to create this construct!

Once we feel good about the design, we can jump more into implementation!

**Comment by scanlonp** [comment]
If you would like to continue working on this, feel free to open a new issue.

**Pull request closed without merging**